operators/src/adapters/raster_time.rs

```diff
@@ -106,15 +113,44 @@ where
         let RasterTimeAdapterProjection {
             source_a,
             source_b,
+            mut query_a_b_fut,
             query_rect,
-            time_end,
             num_spatial_tiles,
             current_spatial_tile,
             mut stream,
             ended,
         } = self.project();
 
-        let next = ready!(stream.as_mut().poll_next(cx));
+        if stream.is_none() {
+            if query_a_b_fut.is_none() {
+                let fut1 = source_a.query(*query_rect);
+                let fut2 = source_b.query(*query_rect);
+                let join = future::join(fut1, fut2);
+
+                query_a_b_fut.set(Some(join));
+            }
+
+            if let Some(queries_fut) = query_a_b_fut.as_mut().as_pin_mut() {
+                let queries = ready!(queries_fut.poll(cx));
+                query_a_b_fut.set(None);
+
+                match queries {
+                    (Ok(stream_a), Ok(stream_b)) => {
+                        stream.set(Some(stream_a.zip(stream_b)));
+                    }
+                    (Err(e), _) | (_, Err(e)) => {
+                        *ended = true;
+                        return Poll::Ready(Some(Err(e)));
+                    }
+                }
+            }
+        }
+
+        let next = if let Some(stream) = stream.as_mut().as_pin_mut() {
+            ready!(stream.poll_next(cx))
+        } else {
+            panic!("stream is None")
```

Comment:
You missed a panic here

Reply:
this ist kind of a MustNotHappenException. If we get here and the stream is None the logic in the code above is wrong. I think we should panic then. I could also maybe us an uwrap on the stream here.

Reply:
Hm, yeah, well, looks like an anti-pattern :smile: 
Maybe we should look into states

Reply:
332c216472eeb34b73e2bd3809877f580b97295d